### DISCUSSION TITLE
two papers with overlapping data

### DISCUSSION THREAD

**Discussion by bendichter** [dandi/helpdesk#9] [Q&A] [answered]
Hi team,

Generally, I create dandisets that are collections of sessions associated with a specific paper. I have encountered my first situation where there are two papers that have overlapping data. I expect this to come up quite a bit as more and more data is published on here and reused. What is the practice for creating dandisets in this case? I can easily upload the data twice, once in each place, but that would duplicate data on the DANDI archive so maybe there is a better solution.

Ben

**Comment by satra** [comment, marked as answer]
DANDI supports this and also allows you to create relationships between dandisets through the metadata. You can upload the same asset to two dandisets, and DANDI won't duplicate data if the same data blob is uploaded. In a future iteration you will simply be able to "copy" an asset from one dandiset to another or construct a dandiset from the result of a search on the archive.

**Reply by bendichter** [reply]
Thanks!

**Reply by kabilar** [reply]
Hi team, has the feature to copy an asset from one dandiset to another been implemented?  I would like to use a few assets (~2tb) from an existing dandiset in a new dandiset, and add additional data that was acquired for the subject to the new dandiset.  I don't currently have a copy of the ~2tb assets locally.  What would be the best strategy for this?  Thank you.

**Reply by yarikoptic** [reply]
API already allows for it, and I believe @satra had a script crafted somewhere for a use case.

**Reply by bendichter** [reply]
I'm out of office but maybe check Satra's gists?

**Reply by yarikoptic** [reply]
yeap -- right there on the top https://gist.github.com/satra/29404d965226e4c99fb48e7502953503

**Reply by kabilar** [reply]
Perfect, thank you all

**Reply by kabilar** [reply]
Hi team,

I was able to link a json file to a new dandiset, but am unable to link a zarr file.  I think I am missing something simple since blobs are handled differently than zarr ([zarr developer docs](https://github.com/dandi/dandi-archive/blob/master/doc/design/zarr-support-3.md)).  

For a zarr file, I changed line 29 in the above [gist script](https://gist.github.com/satra/29404d965226e4c99fb48e7502953503) from `"blob_id": asset.blob` to `"zarr_id": asset.zarr`, but am receiving the error below at line 32.  The error message appears to be getting invoked in the dandiapi [here](https://github.com/dandi/dandi-archive/blob/951b1542c228324a65e45f859180bc198de6333a/dandiapi/api/services/asset/__init__.py#L126-L128).

Error message
```
Error 400 while sending POST request to https://api.dandiarchive.org/api/dandisets/000724/versions/draft/assets/: "The zarr archive belongs to a different dandiset"
```

Entire traceback
```
HTTPError                                 Traceback (most recent call last)
File ~/Desktop/add_asset_to_dandiset.py:32
     31     # this will generate an exception if the path already exists
---> 32     out = da.post(path=ds.api_path + 'versions/draft/assets/', data=data)

File ~/miniconda3/envs/dandi/lib/python3.10/site-packages/dandi/dandiapi.py:308, in RESTFullAPIClient.post(self, path, **kwargs)
    304 def post(self, path: str, **kwargs: Any) -> Any:
    305     """
    306     Convenience method to call `request()` with the 'POST' HTTP method.
    307     """
--> 308     return self.request("POST", path, **kwargs)

File ~/miniconda3/envs/dandi/lib/python3.10/site-packages/dandi/dandiapi.py:274, in RESTFullAPIClient.request(self, method, path, params, data, files, json, headers, json_resp, retry_statuses, retry_if, **kwargs)
    272         raise HTTP404Error(msg, response=result)
    273     else:
--> 274         raise requests.HTTPError(msg, response=result)
    276 if json_resp:
    277     if result.text.strip():

HTTPError: Error 400 while sending POST request to https://api.dandiarchive.org/api/dandisets/000724/versions/draft/assets/: "The zarr archive belongs to a different dandiset"
```

**Reply by yarikoptic** [reply]
eh, yet another "zarrs are special" case I guess. We really need to approach finalizing their design in terms of supporting versioning/publishing and this current 1-to-1  mapping of them to dandisets. attn @satra and @waxlamp

**Reply by kabilar** [reply]
Thank you @yarikoptic.  That is good to know.  For now as a workaround, I will point users of this new dandiset 724 to the zarr assets in the existing dandiset 26.

**Reply by kabilar** [reply]
Continuing the discussion in https://github.com/dandi/dandi-archive/issues/1792.
